### PR TITLE
Fixed rake examples:build

### PR DESCRIPTION
**Pull request by benasher44**
- Fixed `rake examples:build` never calling into `xcodebuild`
- Updated Alamofire example, since it was using an outdated version of swift

In doing #5063, I noticed that travis reported a passing build, even though the point of that PR was to add a failing test case to be fixed in #5049. After some digging, it turns out that you can't switch on an xcode target as defined in [Xcodeproj](/Cocoapods/Xcodeproj), and what was probably meant here was to switch on `project.platform_name`. Once that was fixed, I had some issues to fix with the Alamofire sample project, which required updating it to a modern version of Alamofire that uses a modern version of Swift.

Once this PR is merged, I'll update #5063 so that the failing build is properly reported.

### CODE REVIEW

**Comment by benasher44** [comment]
Added Alamofire as a local pod, since [travis had some trouble](https://travis-ci.org/CocoaPods/CocoaPods/builds/117500238) finding this version for some reason.

**Comment by benasher44** [comment]
I think I need some guidance on what to do next from someone from the CocoaPods team. Now the build appears to fail because the version of Swift is too modern for the version of `xcodebuild` in use. Should I specify xcode7.2 in the .travis.yml? Maybe there's some strategic reason that there is a conditional check in the `examples:build` `rake` task for the Xcode version (to change how we build based on the `xcodebuild` version) that I don't understand. From a project management perspective, it seems easier to just pin to the latest version of GM xcode available from travis and just make sure that the projects work with that one, since that what clients of CocoaPods should be using to submit apps.

**Comment by benasher44** [comment]
Welp, using the xcode7.2 osx image had other issues. Thoughts?

**Comment by benasher44** [comment]
Well aside from making some tweaks to the .travis.yml that you all may or may not want, this branch is now ready to go in terms of just fixing `rake examples:build`

File: Rakefile [hunk 1]
```diff
@@ -266,7 +266,7 @@ begin
             project = Xcodeproj::Project.open(project_path)
             target = project.targets.first
 
-            case target
+            case target.platform_name
```
Comment (segiddins) [inline Rakefile]: please add an `else` case here to raise if theres an unknown platform name, that way this can't fail silently again

**Comment by segiddins** [comment]
:+1: other than the one comment, thanks so much for the PR!

File: Rakefile [hunk 2]
```diff
@@ -275,6 +276,8 @@ begin
               # Specifically build against the simulator SDK so we don't have to deal with code signing.
               simulator_name = major_version > 5 ? 'iPhone 6' : 'iPhone Retina (4-inch)'
               execute_command "xcodebuild -workspace '#{workspace_path}' -scheme '#{scheme_name}' clean build ONLY_ACTIVE_ARCH=NO -destination 'platform=iOS Simulator,name=#{simulator_name}'"
+            else
+              fail "Unknown platform #{platform.to_s}"
```
Comment (segiddins) [inline Rakefile]: `raise "Unknown platform #{platform}"`

**Comment by benasher44** [comment]
Hm. I also noticed it's not building the watchOS example. Gonna try to fix that too.

File: Rakefile [hunk 3]
```diff
@@ -277,7 +277,7 @@ begin
               simulator_name = major_version > 5 ? 'iPhone 6' : 'iPhone Retina (4-inch)'
               execute_command "xcodebuild -workspace '#{workspace_path}' -scheme '#{scheme_name}' clean build ONLY_ACTIVE_ARCH=NO -destination 'platform=iOS Simulator,name=#{simulator_name}'"
             else
-              fail "Unknown platform #{platform.to_s}"
+              raise "Unknown platform #{platform.to_s}"
```
Comment (segiddins) [inline Rakefile]: `#{platform}`, the `.to_s` is redundant and I believe rubocop will complain about it

**Comment by benasher44** [comment]
@segiddins  I changed it to use not specify the destination, and now (after recreating the watchkit extension because configuration was borked) all of the projects build correctly. Not specifying the destination has a few benefits:

- For the watch, different versions of Xcode come with different iOS simulators linked to a watch simulator by default (iPhone 6s in Xcode 7.3), and adding more simulators to link to the watch simulator isn't a configuration that's in an xcscheme or somewhere else that's under version control. So maintaining building the watch sample over time will likely become frustrating.
- Don't need to worry about which platform you're building for, since that info is encoded in the project/scheme.
- `xcodebuild` will just find a destination that works for the scheme for you to accomplish the build.

And, since we're just building the projects and not running tests (i.e `XCTest`) for them, there's really no need to have a destination.

**Comment by benasher44** [comment]
Oh but now I'm reading the note about code-signing that was in the Rakefile, but maybe that's no longer relevant with Xcode 7, so we'll see after this travis run. I'll go back and add destinations if that breaks.

**Comment by segiddins** [comment]
You'll have to revert that last commit - the code signing issue still exists (https://travis-ci.org/CocoaPods/CocoaPods/jobs/120174319#L3291-L3294)

**Comment by benasher44** [comment]
@segiddins dang. Well, there are still good watchOS changes in that commit that are important to make watchOS project build. I'll update the Rakefile soon. I was running into issues getting the destination to be correct when building the watchOS target, so now I think I've found a bug in Xcodeproj. Debugging that now.

**Comment by segiddins** [comment]
:+1: thanks so much again for taking this on!

**Comment by benasher44** [comment]
The issues turned out to be a combination of more bad scheme config in the watch sample and `rake examples:build` attempting to build things in the Pods.xcodeproj, since I guess development pods are automatically added to your main project's schemes. It should be okay now.

**Comment by benasher44** [comment]
This still isn't exercising the watchos case, even though everything should be working now. Working on that.

**Comment by benasher44** [comment]
I regenerated the schemes, and everything looks more normal to me now. It's still never going to switch on a platform that's `:watchos`, which is likely because building the watch app requires building the iOS parts as well, so the first target in the scheme is an iOS target. So, I'm happy with this now. Hope travis brings good news.

**Comment by benasher44** [comment]
As I noted earlier, Xcode 7.3 seems to only come with the iPhone 6s paired to the watch sim, which is why it complained when it went to try to build the watch project with the iPhone 6, so I had to switch to the iPhone 6s.

**Comment by segiddins** [comment]
:+1: but I'd also like either @neonichu or @mrackwitz to also have a look at this before merging

File: Rakefile [hunk 4]
```diff
@@ -261,20 +260,21 @@ begin
...
 
-            case target
+            platform = target.platform_name
+            case platform
             when :osx
               execute_command "xcodebuild -workspace '#{workspace_path}' -scheme '#{scheme_name}' clean build"
             when :ios
-              xcode_version = `xcodebuild -version`.scan(/Xcode (.*)\n/).first.first
-              major_version = xcode_version.split('.').first.to_i
-              # Specifically build against the simulator SDK so we don't have to deal with code signing.
```
Comment (neonichu) [inline Rakefile]: We should retain this comment :)

**Comment by neonichu** [comment]
LGTM (except for the one comment), thanks a lot for the PR @benasher44 :tada:

**Comment by mrackwitz** [comment]
Wow, this was *really* broken. Thanks for a lot for putting in all this effort to make it work again! LGTM :+1:

File: Rakefile (hunk 4 above)
Comment (benasher44) [inline Rakefile]: Woops! That's definitely still relevant. Nice catch!

**Comment by benasher44** [comment]
Well, at least with the 2nd run you can see that the examples to run/pass when execution_output thing isn't flaking.

**Comment by segiddins** [comment]
@benasher44 :rocket: thanks so much!

**Comment by orta** [comment]
Yep - great work @benasher44

**Pull request merged**